Plutarch/Bool.hs

```diff
@@ -35,3 +35,19 @@ pif :: Term s PBool -> Term s a -> Term s a -> Term s a
 pif b case_true case_false = pmatch b $ \case
   PTrue -> case_true
   PFalse -> case_false
+
+-- | Boolean negation for 'PBool' terms.
+pnot :: Term s (PBool :--> PBool)
+pnot = phoistAcyclic $ plam $ \x -> pif x (pcon PFalse) $ pcon PTrue
+
+-- | Lazily evaluated boolean and for 'PBool' terms.
+infixr 3 #&&
+
+(#&&) :: Term s PBool -> Term s PBool -> Term s PBool
+x #&& y = pif x (pif y (pcon PTrue) $ pcon PFalse) $ pcon PFalse
+
+-- | Lazily evaluated boolean or for 'PBool' terms.
+infixr 2 #||
+
+(#||) :: Term s PBool -> Term s PBool -> Term s PBool
+x #|| y = pif x (pcon PTrue) $ pif y (pcon PTrue) $ pcon PFalse
```

Comment:
I wonder whether we could have some way of having Plutarch-level operators.

Reply:
That's certainly something to think about. But I think the best you can do is have a separate function that "lifts" plutarch level functions to let them work as "infix operators" - though that kinda ruins the idea.

I assume we wouldn't need it for `&&` and `||` though.

Reply:
I thought a bit more about the efficiency of this. Do you think something like this would be any better?
```hs
(#&&) :: Term s PBool -> Term s PBool -> Term s PBool
x #&& y = _pand # pdelay x # pdelay y

_pand :: Term s (PDelayed PBool :--> PDelayed PBool :--> PBool)
_pand = phoistAcyclic $ plam
  $ \x y -> pif' # pforce x # pcon PTrue #$ pif' # pforce y # pcon PTrue # pcon PFalse
```

Reply:
Very likely yes for both. You might want to just make it `pand`, since users may want to use it directly. We might also want a strict variant.

Reply:
Done!